Makefile

```diff
@@ -0,0 +1,56 @@
+# Makefile for Harbor-arm project
+#
+# Targets:
+#
+# all: 
+
+BUILDPATH=$(CURDIR)
+HARBOR_MAKEFILE_PATH=$(BUILDPATH)/Makefile
+HARBOR_PHOTON_MAKEFILE_PATH=$(BUILDPATH)/make/photon/Makefile
+HARBOR_PHOTON_CHARTSERVER_COMPILE_PATH=$(BUILDPATH)/make/photon/chartserver/compile.sh
+HARBOR_PHOTON_REGISTRY_DOCKERFILE_PATH=$(BUILDPATH)/make/photon/registry/Dockerfile.binary
+HARBOR_PHOTON_TRIVY-ADAPTER_DOCKERFILE_PATH=$(BUILDPATH)/make/photon/trivy-adapter/Dockerfile.binary
+HARBOR_PHOTON_NOTARY_DOCKERFILE_PATH=$(BUILDPATH)/make/photon/notary/binary.Dockerfile
+HARBOR_PHOTON_PORTAL_DOCKERFILE_PATH=$(BUILDPATH)/make/photon/portal/Dockerfile
+
+SEDCMD=$(shell which sed)
+SEDCMDI=$(SEDCMD) -i
+ifeq ($(shell uname),Darwin)
+    SEDCMDI=$(SEDCMD) -i ''
+endif
+
+_update_makefile:
+	@echo "update goharbor makefile"
+	@$(SEDCMDI) 's/--rm/--rm --env CGO_ENABLED=0 --env GOOS=linux --env GOARCH=arm64/g' $(HARBOR_MAKEFILE_PATH);
+	@$(SEDCMDI) 's/$$(DOCKERBUILD)/docker buildx build --platform linux\/arm64 --progress plain --output=type=registry/g' $(HARBOR_MAKEFILE_PATH)
+
+_update_make_photon_makefile:
+	@echo "update goharbor photon makefile"
+	@$(SEDCMDI) 's/$(DOCKERCMD) build/$(DOCKERCMD) buildx build --platform linux\/arm64 --progress plain --output=type=registry/' $(HARBOR_PHOTON_MAKEFILE_PATH)
+
+_update_photon_chartserver:
+	@echo "update goharbor chartserver compile.sh"
+	@$(SEDCMDI) 's/go build -a/GOOS=linux GOARCH=arm64 CGO_ENABLED=0 go build -a/g' $(HARBOR_PHOTON_CHARTSERVER_COMPILE_PATH)
+
+_update_photon_registry:
+	@echo "update goharbor Dockerfile.binary"
+	@$(SEDCMDI) 's/CGO_ENABLED=0/GOOS=linux GOARCH=arm64 CGO_ENABLED=0/g' $(HARBOR_PHOTON_REGISTRY_DOCKERFILE_PATH)
+
+_update_photon_trivy-adapter:
+	@echo "update goharbor trivy-adapter Dockerfile.binary"
+	@$(SEDCMDI) 's/CGO_ENABLED=0/GOARCH=arm64 CGO_ENABLED=0/g' $(HARBOR_PHOTON_TRIVY-ADAPTER_DOCKERFILE_PATH)
+
+_update_photonn_portal:
+	@echo "update goharbor portal Dockerfile"
+	@$(SEDCMDI) 's/node:15.4.0/--platform=$${BUILDPLATFORM:-linux\/amd64} node:15.4.0/g' $(HARBOR_PHOTON_PORTAL_DOCKERFILE_PATH);
+	@$(SEDCMDI) "s/'node_modules\/@angular\/cli\/bin\/ng'/.\/node_modules\/@angular\/cli\/bin\/ng/g" $(HARBOR_PHOTON_PORTAL_DOCKERFILE_PATH)
+
+
+_update_photon_notary:
+	@echo "update goharbor notary binary.Dockerfile"
+	@$(SEDCMDI) '8 a ENV CGO_ENABLED 0 \nENV GOOS linux \nENV GOARCH arm64' $(HARBOR_PHOTON_NOTARY_DOCKERFILE_PATH)
+
+
+_update_photon_redis:
+
+pre_update: _update_makefile _update_make_photon_makefile _update_photon_chartserver _update_photon_registry _update_photon_trivy-adapter _update_photon_notary _update_photonn_portal
```

Comment:
what's the basic flow? and the entrypoint?

make install？

Reply:
The basic workflow reuses the harbor-core construction process